test/collectBy.js

```diff
@@ -0,0 +1,25 @@
+var R = require('../source');
+var eq = require('./shared/eq');
+
+
+describe('collectBy', function() {
+  it('splits the list into sub-lists according to the grouping function', function() {
+    var meals = [
+      {type: 'breakfast', item: '☕️'},
+      {type: 'lunch', item: '🌯'},
+      {type: 'dinner', item: '🍝'},
+      {type: 'breakfast', item: '🥐'},
+      {type: 'lunch', item: '🍕'}
+    ];
+    eq(R.collectBy(R.prop('type'), meals), [
+      [{type: 'breakfast', item: '☕️'}, {type: 'breakfast', item: '🥐'}],
+      [{type: 'lunch', item: '🌯'}, {type: 'lunch', item: '🍕'}],
+      [{type: 'dinner', item: '🍝'}]
+    ]);
+  });
+
+  it('returns an empty list if given an empty array', function() {
+    eq(R.collectBy(R.prop('x'), []), []);
+  });
+
```

Comment:
I think we might want a test-case with a function that returns numbers.  (Maybe my modulus example above.)

Reply:
Makes sense

Reply:
I am starting to find review conversations harder to follow on GitHub.  I thought I had made a suggestion for a simpler example, but I can't find it now.  The version of this in the test suite seems perfect, but here, something like this might be simpler:

```js
collectBy (n => n % 10, [5, 11, 12, 22, 23, 25, 32, 35, 43])
//=> [[11], [12, 22, 32], [23, 43], [5, 25, 35]]
```

Reply:
I resorted to property-based testing since you `jsverify` is available.

Reply:
That's fine.  I have very mixed feelings about property-based testing.  It's clearly a better way to catch edge cases, and that's the most important facet of unit tests.  But a strong secondary feature of unit tests is to serve as an extra API guide.  When I look to see how a project or a new feature works, I usually look first to the test suite rather than the docs.  That doesn't work nearly as well with property-based testing.